src/service/private.overseas.entity.details.ts

```diff
@@ -0,0 +1,32 @@
+import { Session } from "@companieshouse/node-session-handler";
+import { Request } from "express";
+import { makeApiCallWithRetry } from "./retry.handler.service";
+import { createAndLogErrorRequest, logger } from "../utils/logger";
+import { OverseasEntityExtraDetails } from "@companieshouse/api-sdk-node/dist/services/overseas-entities/types";
+
+export const getPrivateOeDetails = async (
+  req: Request,
+  oeNumber: string,
+): Promise<OverseasEntityExtraDetails | undefined> => {
+  const response = await makeApiCallWithRetry(
+    "overseasEntity",
+    "getOverseasEntityDetails",
+    req,
+    req.session as Session,
+    oeNumber,
+  );
+
+  if (response.httpStatusCode !== 200 && response.httpStatusCode !== 404) {
+    const errorMsg = `Something went wrong fetching company details = ${JSON.stringify(response)}`;
+    throw createAndLogErrorRequest(req, errorMsg);
+  }
+
+  if (response.httpStatusCode === 404) {
+    logger.debugRequest(req, `No company data found for ${oeNumber}`);
```

Comment:
This deserves a error log message too.

Reply:
Similar to the response to the comment on line 20, this is the same as the logging for fetching the company profile data. Any particular reason why they should differ or should company profile also be updated?

Reply:
not too concerned about this one to be fair.